roles/ceph-common/handlers/main.yml

```diff
@@ -1,8 +1,12 @@
 ---
+# upgrading ceph from hammer to jewel will make the restarting failed,
+# this will cause ceph cluster down.
+# this happens when operater really upgrades ceph and forget to set ceph.upgrade2jewel=true
+# so we can't set default value of ceph.upgrade2jewel as false here
 - name: restart ceph mons
   service: name=ceph-mon-all state=restarted
-  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool
+  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool and not ceph.upgrade2jewel|bool
```

Comment:
Perhaps we can use some other run-time fact to discover if this particular deploy has resulted in a ceph package upgrade. We will be using the `upgrade` flag in future releases and may need to restart ceph for config change.